### PR TITLE
Feature/auto disable browser sign in

### PR DESCRIPTION
**Pull request by danfalc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,19 @@
 # chips_automated_testing
 Automated Tests for CHIPS using Selenium 4
 
+### Before running tests
+
+Browser sign in is enabled by default on Companies House Windows machines. However, we cannot run automated tests without disabling this feature. It is necessary to do this through
```
Comment (rsuller) [inline README.md:6]: Some minor changes:

> Browser sign-in is enabled by default on Companies House Windows machines. However, we cannot run automated tests without disabling this feature. It is necessary to do this through 
> the registry, by setting the Edge browserSignIn d-word value to 0. A Powershell script has been written to perform this action and can be run by following the below:
> 
> ( in IntelliJ ) Right-click on browserSignIn.ps1 and select Show in Explorer, alternatively you can navigate to the top directory of the project manually, at the same level as pom.xml
> At this point, you can now right-click and select the option Run with Powershell
> After a few seconds, the script should execute. You may be prompted to accept Powershell making changes to your machine. Click yes to allow this if so
> and set the registry value to 0. You can check this by navigating to the registry editor from the start menu
> 
> From time to time and after restarting your machine, this value can be reset automatically. When running tests, if you see the Microsoft Edge sign-in prompt it is necessary to re-run 
> the Powershell script using the instructions above

File: browserSignIn.ps1 [hunk 2]
```diff
@@ -0,0 +1,36 @@
+ [CmdletBinding()]
```
Comment (rsuller) [inline browserSignIn.ps1:1]: 🤓

**Review by rsuller — APPROVED** [review]

File: browserSignIn.ps1 [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        Try {
+            Get-ItemPropertyValue -Path 'HKLM:\SOFTWARE\Policies\Microsoft\Edge\' -Name browserSignIn -ErrorAction Stop
+        }
+        Catch {
+            Throw "browserSignIn not found in registry";
+        }
+    }
+}
+
+# Instructions from https://chromedriver.chromium.org/downloads/version-selection
```
Comment (flisowski) [inline browserSignIn.ps1:17]: Don't think those 2 comment lines go here?

**Comment by flisowski** [comment]
This script must be rapid and execute in ms - would it be an option to have it run pre-tests? Perhaps called from pr-execution hooks since it doesn't take time? Thinking about what you said about the value of that registry entry resetting itself every now and then.

File: browserSignIn.ps1 (hunk 3 above)
Comment (danfalc) [reply browserSignIn.ps1]: resolved

**Comment by danfalc** [comment]
> This script must be rapid and execute in ms - would it be an option to have it run pre-tests? Perhaps called from pre-execution hooks since it doesn't take time? Thinking about what you said about the value of that registry entry resetting itself every now and then.


Tried and tested execution in hooks. It worked on my machine but on Suller's mac it brings up a prompt that has to be accepted manually each time a test is run. So decided to take it out of the hooks and have it as a script that can be run ad hoc

**Pull request merged**